### PR TITLE
Make beat.reference.yml optional.

### PR DESCRIPTION
**Pull request by simitt**
Add ENV variable to allow removal of `beat.reference.yml`.

### CODE REVIEW

**Comment by ruflin** [comment]
jenkins, test this

File: libbeat/scripts/Makefile [hunk 1]
```diff
@@ -301,12 +302,18 @@ update: python-env collect
 
 	@# Update config files.
 	@cat _meta/beat.yml ${ES_BEATS}/libbeat/_meta/config.yml | sed -e "s/beatname/${BEAT_NAME}/g;s/beat-index-prefix/${BEAT_INDEX_PREFIX}/g" > ${BEAT_NAME}.yml
+	@chmod 0640 ${BEAT_NAME}.yml
```
[the remark this answers is not in the document]
Comment (simitt) [reply libbeat/scripts/Makefile]: That behavior has been there before, see https://github.com/elastic/beats/pull/7092/files/04a6075f887fc3a594791baa0c27bb18c4f7c9ba#diff-d2f9f6e7040050cd09938355f7d358a6L309

I just moved it up so it is always done, independently of having a reference file or not.

**Comment by simitt** [comment]
Yes exactly, we would like to combine to one `yml` file. I think we talked about this a while ago.

**Comment by ruflin** [comment]
I broke the packaging with an other PR. Hope to get a fix in today. Can you rebase afterwards again?

**Comment by ruflin** [comment]
@simitt Other PR's are in, can you rebase?

**Comment by simitt** [comment]
@ruflin I had `make package-all` run which finished without errors.

**Comment by ruflin** [comment]
@simitt Thanks a lot for testing it. Merged. Now lets keep an eye on the next packaging run.

**Pull request merged**